packages/tx-construction/src/tx-builder/TxBuilder.ts

```diff
@@ -289,6 +289,8 @@ export class GenericTxBuilder implements TxBuilder {
                 auxiliaryData,
                 ownAddresses,
                 signingContext: {
+                  // TODO: same as wallet.governance.getPubDRepKey?
```

Comment:
`dRepPublicKey` is required in the `checkDrepPublicKeyAgainstCredential` to map out `RegisterDelegateRepresentative, UnregisterDelegateRepresentative` and `UpdateDelegateRepresentative certigicates`, but i'm not sure it is passed all the way down to sign tx call
stack trace is 
```
          at LedgerKeyAgent.signTransaction (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/hardware-ledger/src/LedgerKeyAgent.ts:539:13)
          at Function.getSignatures (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/key-management/src/util/createWitnesser.ts:74:39)
          at Bip32Ed25519Witnesser.witness (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/key-management/src/util/createWitnesser.ts:38:39)
          at Object.sign (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/tx-construction/src/tx-builder/TxBuilder.ts:325:47)
          at LazyTxSigner.sign (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/tx-construction/src/tx-builder/TxBuilder.ts:91:25)
          at Object.<anonymous> (/Users/vitaliivozniak/Documents/projects/iohk/cardano-js-sdk/packages/wallet/test/hardware/ledger/LedgerKeyAgent.test.ts:547:18)
```

@mirceahasegan would you be able to suggest here?

Reply:
There can be only one drepPublicKey, so in the InMemoryKeyAgent we derive it [before calculating](https://github.com/input-output-hk/cardano-js-sdk/blob/e7dbf35c224ccbb8bbd75a3cd031c8f4c1288a38/packages/key-management/src/InMemoryKeyAgent.ts#L133).
I assume the same should be done for the Ledger key agent